lib/maths/time_series/CTimeSeriesModel.cc

```diff
@@ -1075,13 +1076,25 @@ bool CUnivariateTimeSeriesModel::uncorrelatedProbability(
         result.s_AnomalyScoreExplanation.s_UpperConfidenceBound = interval[2][0];
     }
 
+    result.s_AnomalyScoreExplanation.s_MultimodalDistribution = this->multimodalPrior();
+    LOG_DEBUG(<< "Multimodel distribution: "
+              << result.s_AnomalyScoreExplanation.s_MultimodalDistribution);
+
     result.s_Probability = pOverall;
     result.s_FeatureProbabilities = std::move(featureProbabilities);
     result.s_Tail = {tail};
 
     return true;
 }
 
+bool CUnivariateTimeSeriesModel::multimodalPrior() const {
+    LOG_DEBUG(<< "Residual model type " << m_ResidualModel->type());
+    if (m_ResidualModel->type() == common::CPrior::E_OneOfN) {
```

Comment:
So I can see why you did it like this. My inclination would be we have an `bool isSelectedModelMultimodal()` defined on `CPrior` which returns false and override it in `COneOfOneN` and ditch this function altogether.

Reply:
I see how this can simplify the code. I updated the code according to your suggestion.